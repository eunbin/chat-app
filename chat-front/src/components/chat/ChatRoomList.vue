<template>
  <v-card>
    <v-list
      two-line>
      <template v-for="(item, index) in list">
        <v-list-tile
          :key="item.index"
          @click="onSelectChatRoom(item)"
        >
          <v-list-tile-content>
            <v-list-tile-title>{{ item.title }}</v-list-tile-title>
            <v-list-tile-sub-title class="text-xs-right">
            </v-list-tile-sub-title>
          </v-list-tile-content>
        </v-list-tile>
        <v-divider
          v-if="index < list.length - 1"
          :key="index">
        </v-divider>
      </template>
    </v-list>
  </v-card>
</template>

<script>
export default {
  name: 'ChatRoomList',
  data () {
    return {
    }
  },
  props: {
    list: {
      type: Array,
      default: () => []
    }
  },
  methods: {
    onSelectChatRoom (item) {
      this.$emit('select-room', item)
    }
  }
}
</script>
