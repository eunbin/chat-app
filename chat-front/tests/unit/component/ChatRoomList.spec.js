import { mount } from '@vue/test-utils'
import '../test-setup'
import ChatRoomList from '@/components/chat/ChatRoomList.vue'

describe('ChatRoomList.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(ChatRoomList)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('snapshot check with items', () => {
    const list = [{
      roomId: 1,
      title: 'Hello Chat Room',
      members: []
    }]
    const wrapper = mount(ChatRoomList, {
      propsData: { list }
    })

    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('given room list when click room then equal item data', () => {
    const list = [{
      roomId: 1,
      title: 'Hello Chat Room',
      members: []
    }]
    const wrapper = mount(ChatRoomList, {
      propsData: { list }
    })

    wrapper.find('.v-list__tile').trigger('click')
    expect(wrapper.emitted()['select-room'][0][0]).toEqual(list[0])
  })
})
