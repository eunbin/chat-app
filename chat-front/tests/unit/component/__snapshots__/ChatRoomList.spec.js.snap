// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatRoomList.vue snapshot 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-list v-list--two-line theme--light"
    role="list"
  />
</div>
`;

exports[`ChatRoomList.vue snapshot check with items 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-list v-list--two-line theme--light"
    role="list"
  >
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Hello Chat Room
          </div>
           
          <div
            class="v-list__tile__sub-title text-xs-right"
          />
        </div>
      </a>
    </div>
     
    <!---->
  </div>
</div>
`;
